src/top.rs

```diff
@@ -440,7 +482,55 @@ impl Link {
     }
 
     fn explicit_partition(&mut self) {
-        self.state = State::ExplicitPartition;
+        self.state = State::ExplicitPartitionBothWays;
+    }
+
+    fn explicit_partition_a_to_b(&mut self) {
```

Comment:
We'll also need test coverage over these state transitions.

Reply:
I added an exhaustive test that all 6 operations that affect partitioning do work as expected in all 6! (=720) combinations.

This gives some peace of mind, but it takes a while to run in debug-mode (about 8 seconds) . One possibility would be to just disable this test for debug-builds. Another possibility would be to not run all 720 combinations.

Reply:
I think we could limit the number of cases down to still be pretty fast, but also give us confidence in the implementation. You could also use a lower level test on this module that doesn't need all the packets to move around or all the host scheduling.

Reply:
Done!